Fake clock does not handle negative durations to After() correctly

# Description
`fakeClock` does not handle negative durations passed to `After()` correctly.

This looks very similar to https://github.com/jonboulle/clockwork/issues/19

# Steps to reproduce
```go
func TestNegativeTimer(t *testing.T) {
	fakeClock := clockwork.NewFakeClock()
	timer := time.NewTimer(2 * time.Second)
	defer timer.Stop()
	select {
	case <-fakeClock.After(-1 * time.Second):
	// Success
	case <-timer.C:
		t.Fatalf("Timed out!")
	}
}
```
Output:
```
$ go test -v -run TestNegativeTimer
=== RUN   TestNegativeTimer
--- FAIL: TestNegativeTimer (2.00s)
    negativetimer_test.go:18: Timed out!
FAIL
exit status 1
```

--- a/clockwork.go
+++ b/clockwork.go
@@ -97,7 +97,7 @@ func (fc *fakeClock) After(d time.Duration) <-chan time.Time {
 	defer fc.l.Unlock()
 	now := fc.time
 	done := make(chan time.Time, 1)
-	if d.Nanoseconds() == 0 {
+	if d.Nanoseconds() <= 0 {
 		// special case - trigger immediately
 		done <- now
 	} else {

--- a/clockwork_test.go
+++ b/clockwork_test.go
@@ -9,6 +9,13 @@ import (
 func TestFakeClockAfter(t *testing.T) {
 	fc := &fakeClock{}
 
+	neg := fc.After(-1)
+	select {
+	case <-neg:
+	default:
+		t.Errorf("negative did not return!")
+	}
+
 	zero := fc.After(0)
 	select {
 	case <-zero:

